tools/cmake/toolchains/GCC_ARM.cmake

```diff
@@ -67,23 +67,23 @@ function(mbed_set_toolchain_options target)
 endfunction()
 
 # GCC ARM requires preprecessing linker script, execute generators to get definitions needed for
-# this step - linker options and compile definitions 
-function(mbed_generate_gcc_options_for_linker target definitions_file linker_options_file)
+# this step - compile definitions
+function(mbed_generate_gcc_options_for_linker target definitions_file)
     set(_compile_definitions 
         "$<TARGET_PROPERTY:${target},COMPILE_DEFINITIONS>"
     )
-
-    set(_linker_options
-        "$<TARGET_PROPERTY:${target},LINK_OPTIONS>"
+    # We filter -D to the linker script, as only simple ones are passed (value based)
+    # To properly fix, we will need to replace all these the same way as Ninja or other
+    # generator do
+    set(_compile_definitions
+       "$<FILTER:${_compile_definitions},EXCLUDE,.*[ ={ | =" | =\( ].*[ } | " | )]>"
```

Comment:
Okay, as it's a temporary (if not pretty) workaround.

From my suggestion in #13765 

> What I mean is, have a predefined list of macros (MBED_APP_SIZE, MBED_APP_START, MBED_BOOT_STACK_SIZE, etc.) and filter out any macros not on the list (instead of filtering out spaces, brackets, etc.). Alternatively, we could explicitly query those on the list. The same static list could be used for any ld files - no need to look into the ld file we process.
>
> (Just a suggestion - not sure how feasible it is.)

Reply:
Tried to come up with a list
```
$ find targets/ -name "*.ld" | gxargs -l1 grep --only-matching 'MBED_[A-Z0-9_]*' | sort -u
MBED_APP_COMPILE
MBED_APP_LENGTH
MBED_APP_SIZE
MBED_APP_START
MBED_BOOT_STACK_SIZE
MBED_CONF_TARGET_BOOT_STACK_SIZE
MBED_CRASH_REPORT_RAM_SIZE
MBED_CRASH_REPORT_RAM_START
MBED_DATA_SHARED_SIZE
MBED_INTERRUPTS_SIZE
MBED_PUBLIC_RAM_SIZE
MBED_PUBLIC_RAM_START
MBED_RAM0_SIZE
MBED_RAM0_START
MBED_RAM1_SIZE
MBED_RAM1_START
MBED_RAM_APP_SIZE
MBED_RAM_APP_START
MBED_RAM_SIZE
MBED_RAM_START
MBED_ROM_SIZE
MBED_ROM_START
MBED_VECTTABLE_RAM_SIZE
```
Probably not everything is needed...

Reply:
We could do what ninja does and fix offending macros. I would need more time to find a better regex magic. CMAke provides string replace with regex. I'll create a ticket for this to look at it later. We can meanwhile use bootloader or other features with this approach.

Reply:
> We could do what ninja does and fix offending macros. I would need more time to find a better regex magic.

That's an option. (I had a quick look into the CMake source code, but didn't have time to commit to finding it...)

Reply:
created IOTMAINTAIN-262